pandas/tests/io/test_sql.py

```diff
@@ -1299,8 +1295,13 @@ class _TestSQLApiEngine(SQLAlchemyMixIn, _TestSQLApi):
     flavor = "sqlite"
     mode = "sqlalchemy"
 
-    def connect(self):
-        return sqlalchemy.create_engine("sqlite:///:memory:")
+    @classmethod
+    def setup_class(cls):
+        cls.engine = sqlalchemy.create_engine("sqlite:///:memory:")
+
+    @classmethod
+    def teardown_class(cls):
+        cls.engine.dispose()
```

Comment:
Shouldn't this still be in the `SQLAlchemyMixIn` so that the other test classes that inherit this also dispose the engine?

Reply:
Agreed, I am changing this.